test/gurka/test_ferry_connections.cc

```diff
@@ -128,44 +150,35 @@ TEST(Standalone, TruckFerryDuration) {
-                                                        "unclassified",   "service",
-                                                        "secondary_link", "tertiary_link"};
-  for (const auto& cls : reclassifiable_ways) {
-    std::map<std::string, std::string> desc = {{"highway", cls}, {"name", ""}};
-    baldr::RoadClass bclass, gclass;
-    std::tie(bclass, gclass) = ReclassifyFerryConnectionEdge(desc);
-    EXPECT_EQ(bclass, baldr::RoadClass::kPrimary);
-    EXPECT_EQ(gclass, baldr::RoadClass::kPrimary);
-  }
-}
-
-TEST(Standalone, DoNotReclassifyFerryConnection) {
+TEST_F(FerryTest, DoNotReclassifyFerryConnection) {
   // roads with these values of 'highway' tag do not participate in search for ferry connection so
   // edge class remains low
   const std::vector<std::string> not_reclassifiable_ways = {"track", "living_street"};
   for (const auto& cls : not_reclassifiable_ways) {
-    std::map<std::string, std::string> desc = {{"highway", cls}, {"name", ""}};
-    baldr::RoadClass bclass, gclass;
-    std::tie(bclass, gclass) = ReclassifyFerryConnectionEdge(desc);
-    EXPECT_GT(bclass, baldr::RoadClass::kPrimary);
-    EXPECT_GT(gclass, baldr::RoadClass::kPrimary);
+    EXPECT_FALSE(edges_were_reclassified({{"highway", cls}}));
   }
 
   // roads with these values of 'service' tag do not participate in search for ferry connection so
   // edge class remains low
   const std::vector<std::string> not_reclassifiable_use = {"parking_aisle", "driveway", "alley",
                                                            "emergency_access", "drive-through"};
   for (const auto& use : not_reclassifiable_use) {
-    std::map<std::string, std::string> desc = {{"highway", "tertiary"},
-                                               {"service", use},
-                                               {"name", ""}};
-    baldr::RoadClass bclass, gclass;
-    std::tie(bclass, gclass) = ReclassifyFerryConnectionEdge(desc);
-    EXPECT_GT(bclass, baldr::RoadClass::kPrimary);
-    EXPECT_GT(gclass, baldr::RoadClass::kPrimary);
+    std::map<std::string, std::string> desc = {{"highway", "service"}, {"service", use}};
+    EXPECT_FALSE(edges_were_reclassified(desc));
   }
 }
+
+TEST_P(FerryTest, ReclassifyFerryConnection) {
+  // for these values of 'highway' tag edge class is upgraded in order to connect ferry to a
+  // high-class road
+  const std::vector<std::string> reclassifiable_ways = {"secondary",      "tertiary",
+                                                        "unclassified",   "service",
+                                                        "secondary_link", "tertiary_link"};
+  for (const auto& cls : reclassifiable_ways) {
+    EXPECT_TRUE(edges_were_reclassified({{"highway", cls}}, GetParam()));
+  }
+}
+
+INSTANTIATE_TEST_SUITE_P(FerryConnectionTest,
+                         FerryTest,
+                         ::testing::Values("motorcar", "hgv", "moped", "motorcycle", "taxi", "bus"));
```

Comment:
this new test is the only using the parameterization and tests all highway types and all vehicular modes